python/ray/rllib/evaluation/tf_policy_graph.py

```diff
@@ -128,9 +128,10 @@ def __init__(self,
             self._stats_fetches = {}
 
         self._optimizer = self.optimizer()
-        self._grads_and_vars = [(g, v)
-                                for (g, v) in self.gradients(self._optimizer)
-                                if g is not None]
+        self._grads_and_vars = [
+            (g, v) for (g, v) in self.gradients(self._optimizer, self._loss)
```

Comment:
Should we break the gradient computation and application into `self.compute_gradients` and `self.apply_gradients`, like is done in the tf optimizers?

Reply:
I added an build_apply_op() function here which can be overriden as well. Though, it is a bit confusing now since there is also the build_apply_gradients() op below which builds up the actual session run.

Reply:
Nice!